### ISSUE TITLE
Deployment to Cloudflare Pages broken with new Nuxt 3.8 app manifest feature

### ISSUE THREAD

**Issue by AaronBeaudoin** [unjs/nitro#1844]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.18.2`
- Nuxt Version:     `3.8.0`
- CLI Version:      `3.9.1`
- Nitro Version:    `2.7.0`
- Package Manager:  `npm@9.8.1`
- Builder:          `-`
- User Config:      `nitro`
- Runtime Modules:  `-`
- Build Modules:    `-`

### Reproduction

https://github.com/AaronBeaudoin/nuxt-3-issue-1

### Describe the bug

In Nuxt 3.8 a new feature was introduced which adds a new `/_nuxt/builds/meta/<buildId>.json` in the build output directory. This causes the following two new entries to be added to `nitro.options.publicAssets`:

```js
{
  ...
  publicAssets: [
    ...
    {
      dir: '/Users/<me>/Projects/test/test-nuxt-3/.nuxt/manifest/meta',
      maxAge: 31536000,
      baseURL: '/_nuxt/builds/meta',
      fallthrough: false
    },
    {
      dir: '/Users/<me>/Projects/test/test-nuxt-3/.nuxt/manifest',
      maxAge: 1,
      baseURL: '/_nuxt/builds',
      fallthrough: false
    }
  ]
}
```

The Cloudflare Pages preset then "copies" these over when generating a `_routes.json` file:

```js
{
  "version": 1,
  "include": [
    "/*"
  ],
  "exclude": [
    "/_nuxt/*",
    "/_nuxt/builds/*",
    "/_nuxt/builds/meta/*",
    ...
  ]
}
```

This results in the following error during deployment:

> Error: Failed to publish your Function. Got error: Error 8000057: Overlapping rules in `_routes.json` are not allowed. Rule "/_nuxt/builds/\*" is overlapped by "/_nuxt/\*". Remove one of the rules to continue.

I originally created https://github.com/nuxt/nuxt/issues/23796 when I discovered this issue, but according to the maintainer who responded it is Nitro's responsibility to ensure that the overlapping URLs created in `nitro.options.publicAssets` are correctly handled when creating `_routes.json`.

### Additional context

For now this issue can be avoided with `experimental: { appManifest: false }` in `nuxt.config.ts`. However, this issue will come back again if any other overlapping URLs ever make it into `nitro.options.publicAssets`.

**Comment by Atinux** [comment]
In the meantime as a workaround you can set in your `nuxt.config.ts`:

```ts
export default defineNuxtConfig({
  experimental: { appManifest: false }
})
```

**Comment by favna** [comment]
@Atinux This was already mentioned in the posts' "Additional context", along with its downsides. Commenting the same again separately may lead to people incorrectly assuming it's a good long-term solution.

**Comment by DavidDeSloovere** [comment]
You know who he is right? I think he can answer whatever he wants 😁

**Comment by Atinux** [comment]
@favna read too fast sorry about this, deleted the comment to avoid any downsides.

**Comment by pi0** [comment]
The fix is released on [nitro@2.7.2](https://github.com/unjs/nitro/releases/tag/v2.7.2)
